ag_data/models.py

```diff
@@ -71,8 +71,30 @@ class AGMeasurement(models.Model):
 
 
 class AGActiveEvent(models.Model):
+    """Stores the currently Active event which must refer to one of the instance in AGEvent table
+    """
+
     agevent = models.ForeignKey(AGEvent, null=True, on_delete=models.SET_NULL)
 
+    def save(self, *args, **kwargs):
+        # check if the inserted event is valid
+        if not AGEvent.objects.filter(pk=self.pk).exists():
+            pass
+
+        # Make sure there's at most one record in table.
+        objs = AGActiveEvent.objects.all()
+        if len(objs) == 1:
+            obj = objs[0]
+            if obj.pk == self.pk:
+                pass
```

Comment:
I feel like this if statement should be `if obj.pk != self.pk: AGActiveEvent.objects.all.delete() ...`